src/transport/FabricTable.cpp

```diff
@@ -337,18 +350,21 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & opCertArray, Validatio
     // It confirms that the certs link correctly (noc -> icac -> rcac), and have been correctly signed.
     ReturnErrorOnFailure(certificates.FindValidCert(nocSubjectDN, nocSubjectKeyId, context, &resultCert));
 
-    ReturnErrorOnFailure(ExtractPeerIdFromOpCert(certificates.GetCertSet()[nocCertIndex], &nocPeerId));
+    UncompressedPeerId uncompressedPeerId;
+    ReturnErrorOnFailure(ExtractPeerIdFromOpCert(certificates.GetCertSet()[nocCertIndex], &uncompressedPeerId));
 
     if (!icac.empty())
     {
         FabricId icacFabric;
         if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR)
         {
-            ReturnErrorCodeIf(icacFabric != nocPeerId.GetFabricId(), CHIP_ERROR_INVALID_CASE_PARAMETER);
+            ReturnErrorCodeIf(icacFabric != uncompressedPeerId.GetUncompressedFabricId(), CHIP_ERROR_INVALID_CASE_PARAMETER);
```

Comment:
This changes the meaning of the check.

It's possible ICAC doesn't have fabric ID attribute. I think the prior behavior would have yielded 0, and same for the default constructed FabricId (unless there was a bug....). Overall, this check must only be done if the ICAC does indeed have a chip-fabric-id. RDN in its subject